changelog.d/22079_correct_filter_condition_via_vector_generate.fix.md

```diff
@@ -0,0 +1,4 @@
+What: Fixed a typo in the condition of the filter component created via `vector generate`.
+How: Replaced the assignment (`=`) with the equality (`==`) operator.
```

Comment:
```suggestion
The `filter` transform now generates a more accurate config when generated via `vector generate` by using a comparison rather than an assignment.
```

Updated to match our changelog style.